### ISSUE TITLE
Up/down keyboard navigation jumps horizontally

### ISSUE THREAD

**Issue by mcsf** [WordPress/gutenberg#44753]
See also: ~#53416~ https://github.com/WordPress/gutenberg/issues/44055

### Step-by-step reproduction instructions

**The horizontal caret position is not properly preserved across paragraph jumps.** All three browsers behave differently. See the attached videos, in which I place the caret at the end of the last paragraph and then just press <kbd>Up</kbd> repeatedly.

Chrome seems to be the best-behaved browser so far:

https://user-images.githubusercontent.com/150562/194362744-0935798a-a26f-409a-b15b-5cf6af2d977a.mov

Firefox gets it almost right, but then note the jump to "placeat" and then to "maxime":

https://user-images.githubusercontent.com/150562/194362767-b1863b06-00cd-4670-b2e9-c942f07c004b.mov

Safari is all over the place, seemingly always setting the caret's horizontal component to match the end of the curernt paragraph:

https://user-images.githubusercontent.com/150562/194362776-a1c19614-2c4c-4d62-9403-13e16de3eebd.mov

### Environment info

Gutenberg trunk, etc.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by ellatrix** [comment]
@mcsf Could you create separate issues for these? :)

**Comment by ellatrix** [comment]
I don't think there's anything we can do here, except for using the wrapper `contenteditable` trick. Safari gives us a wrong range for `document.caretRangeFromPoint( x, y )`, so 🤷‍♀️

**Comment by mcsf** [comment]
> @mcsf Could you create separate issues for these? :)

Done: I extracted the second issue Into #53416 and kept the first one here.

**Comment by bph** [comment]
@mcsf I am getting confused. Could you please verify if this issue was solved with?

- #53454

If not, what's the likelihood this issue is tackled for the upcoming 6.4 release?

**Comment by mcsf** [comment]
> Could you please verify if this issue was solved with?

No, the issue is still present. #53454 wasn't really concerned with it.

> If not, what's the likelihood this issue is tackled for the upcoming 6.4 release?

As Ella says, fixing this requires reimplementing caret movement with the `contentEditable` trick. Considering all the little things we'd like to get in 6.4, I would say it's more likely that this issue will **not** be fixed by then.

**Comment by bph** [comment]
Thank you so much @mcsf - I moved this issue to the Punt to 6.5 column on the WordPress 6.4 board.

**Comment by annezazu** [comment]
[After a review](https://wordpress.slack.com/archives/C02QB2JS7/p1707152945370159) by core editor triage leads and core editor tech leads, this has been removed from the board for 6.5 consideration.

**Comment by mcsf** [comment]
No longer seems to be a problem.
